invenio_deposit/api.py

```diff
@@ -211,12 +218,16 @@ def process_files(data):
 
             with process_files(data) as data:
                 record = Record.create(data, id_=id_)
+
         else:  # Update after edit
+            schema = self['$schema']
             record_pid, record = self.fetch_published()
-            # TODO add support for patching
-            assert record.revision_id == self['_deposit']['pid']['revision_id']
+            if record.revision_id == self['_deposit']['pid']['revision_id']:
+                data = dict(self.dumps())
+            else:
+                data = self.merge_with_published()
 
-            data = dict(self.dumps())
+            data['$schema'] = schema
```

Comment:
> Please note that we need to take care about $schema and _deposit fields.

Did you say to take care about $schema and I though I should preserve like you did for `_deposit` field here: https://github.com/inveniosoftware/invenio-deposit/blob/master/invenio_deposit/api.py#L306-L308
Should I do in a different way?